pyiceberg/io/pyarrow.py

```diff
@@ -1068,8 +1068,17 @@ def primitive(self, primitive: pa.DataType) -> PrimitiveType:
             return StringType()
         elif pa.types.is_date32(primitive):
             return DateType()
-        elif isinstance(primitive, pa.Time64Type) and primitive.unit == "us":
+        elif isinstance(primitive, pa.Time64Type) and (primitive.unit == "us" or primitive.unit == "ns"):
+            primitive = cast(pa.TimestampType, primitive)
```

Comment:
`Time64Type` and `TimestampType` and different, dont need to cast here

Reply:
https://github.com/zaryab-ali/iceberg-python/blob/5c3483b25d8655067930bb8ffe938a6296c6ead6/pyiceberg/types.py#L656